### PR TITLE
Feature/json serializable coordinate

### PR DESCRIPTION
**Pull request by TomSimons**
Added attributes for JSON serialization of Coordinate class if Newtonsoft.JSON is present in the project.
Added JSON constructor for the Coordinate class.

### CODE REVIEW

File: Runtime/Scripts/Coordinate.cs [hunk 1]
```diff
@@ -31,10 +34,12 @@ public struct Coordinate
         /// The CoordinateSystem is defined as an int and not as CoordinateSystem enum so that third-parties can
         /// add their own EPSG conversions that are not (yet) included in the enum.
         /// </remarks>
-        /// 
+        ///
+#if NEWTONSOFT
+        [JsonProperty]
+#endif
+        public int CoordinateSystem { get; }
```
Comment (mvriel) [inline Runtime/Scripts/Coordinate.cs:41]: get without set? Is this a C# construct or am I missing a `private set`?

File: Runtime/Scripts/Coordinate.cs [hunk 2]
```diff
@@ -109,7 +123,19 @@ public Coordinate(CoordinateSystem coordinateSystem, params double[] points)
...
             extraLattitudeRotation = 0;
         }
-
+        
+#if NEWTONSOFT
+        [JsonConstructor]
+#endif
+        public Coordinate(int coordinateSystem, double[] points, double extraLongitudeRotation, double extraLatitudeRotation)
+        {
+            converter = null;
```
Comment (mvriel) [inline Runtime/Scripts/Coordinate.cs:132]: Can we immediately do `CoordinateSystems.operators[(CoordinateSystem)CoordinateSystem];` here to save us some work in the properties? Or does this mess with the order of things?

File: Runtime/Scripts/Coordinate.cs [hunk 3]
```diff
@@ -109,7 +123,19 @@ public Coordinate(CoordinateSystem coordinateSystem, params double[] points)
...
+            CoordinateSystem = coordinateSystem;
+            Points = points;
+            this.extraLongitudeRotation = extraLongitudeRotation;
+            this.extraLattitudeRotation = extraLatitudeRotation;
+        }
+        
         public Coordinate(int coordinateSystem, params double[] points)
         {
             converter = null;
```
Comment (mvriel) [inline Runtime/Scripts/Coordinate.cs:141]: If https://github.com/Netherlands3D/Coordinates/pull/16/files#r1697071867 works.. could we do that here too to be able to remove all null checks? Perhaps we need to check with @mvangog why this is like this

File: Runtime/Scripts/Coordinate.cs (hunk 1 above)
Comment (mvriel) [reply Runtime/Scripts/Coordinate.cs:41]: Tom already answered this, it is indeed a C# trick to get a read only property that is still serializable. When the setter is omitted, you can set it once in the constructor. Similar to a readonly field. But due to the public getter and the JsonConstructor, NewtonSoft can still handle this

**Review by mvriel — APPROVED** [review]

File: Runtime/Scripts/Coordinate.cs (hunk 3 above)
Comment (TomSimons) [reply Runtime/Scripts/Coordinate.cs:141]: I'll leave that with @mvangog, it can be another pr

**Pull request merged**